pace-util/pace/util/quantity.py

```diff
@@ -284,6 +284,7 @@ def __init__(
         else:
             extent = tuple(extent)
 
+        self._dims = dims
         if isinstance(data, (int, float, list)):
             data = np.asarray(data)
         elif gt4py is not None and isinstance(data, gt4py.storage.storage.Storage):
```

Comment:
I might suggest not merging this, but pointing it out so you're aware - there's a bug here where if someone passes a list of floats to initialize the quantity with a z-dimension, and includes a gt4py backend, the storage attribute won't work when you'd expect it should. Since we never initialize quantities this way (perhaps the list option should be removed altogether), this is not an issue.

If you merge this suggestion it might cause unforeseen test failures, so it's really up to you.

```suggestion
        if gt4py is not None and isinstance(data, gt4py.storage.storage.Storage):
```